maintenance/wikia/uncycloUsersMigrator.php

```diff
@@ -0,0 +1,695 @@
+	 * @param string $since MW timestamp string
+	 * @param int $since edits count
+	 */
+	protected function getEditsCountAfter( User $user, $since ) {
+		return $this->getUncycloDB()->selectField(
+			'revision',
+			'count(*)',
+			[
+				'rev_user' => $user->getId(),
+				sprintf( 'rev_timestamp > "%s"', $since )
+			],
+			__METHOD__
+		);
+	}
+
+	/**
+	 * Updates given column (userid_column or username_column) with a new value
+	 *
+	 * @param string $fieldType
+	 * @param string|int $oldValue
+	 * @param string|int $newValue
+	 */
+	protected function updateTables( $fieldType, $oldValue, $newValue ) {
+		$dbw = $this->getUncycloDB(DB_MASTER);
+
+		foreach( self::$mTableRenameRules as $entry ) {
+			// a single entry:
+			// array( 'table' => 'archive', 'userid_column' => 'ar_user', 'username_column' => 'ar_user_text' ),
+			if ( is_null( $entry[ $fieldType ] ) ) {
+				continue;
+			}
+
+			$columnName = $entry[ $fieldType ];
+			$tableName = $entry[ 'table' ];
+
+			if ( !$dbw->tableExists( $tableName ) ) {
+				$this->err( 'Table does not exist', [ 'tableName' => $tableName ] );
+				continue;
+			}
+
+			$dbw->update(
+				$entry[ 'table' ],
+				[ $columnName => $newValue ], // SET
+				[ $columnName => $oldValue ], // WHERE
+				__METHOD__
+			);
+		}
+	}
+
+	/**
+	 * Change uncyclo user ID
+	 *
+	 * Updates fields with user ID only!
+	 *
+	 * @param User $user uncyclo account to migrate
+	 * @param $newUserId new user ID
+	 * @throws Exception
+	 */
+	protected function doChangeUncycloUserId( User $user, $newUserId ) {
+		// "OptiPest" and "MrFlashlight" accounts have the same ID on uncyclo and shared DB
+		if ( $newUserId == $user->getId() ) {
+			return;
+		}
+
+		// don't touch uncyclo users in "global users only" mode
+		if ( $this->onlyRenameGlobalUsers ) {
+			return;
+		}
+
+		// check if there's the uncyclo user with the ID from shared DB
+		$whoIs = User::whoIs( $newUserId );
+
+		if ( $whoIs !== false ) {
+			$this->output( sprintf( "\nCan't change the ID of %s - clashes with %s!\n", $user->getName(), $whoIs ) );
+
+			if (!$this->isDryRun) {
+				throw new UncycloUsersMigratorException(sprintf('IDs clash for %s (new ID taken from shared DB would be #%d)', $user->getName(), $newUserId));
+			}
+		}
+
+		if ( $this->isDryRun ) {
+			return;
+		}
+
+		// update user_id in MW tables
+		$this->updateTables( self::UPDATE_TABLE_USER_ID, $user->getId(), $newUserId );
+	}
+
+	/**
+	 * Rename uncyclopedia user
+	 *
+	 * Updates fields with user name only!
+	 *
+	 * @param User $uncycloUser
+	 * @return User user object with a changed name
+	 */
+	protected function doRenameUncycloUser( User $uncycloUser ) {
+		// don't touch uncyclo users in "global users only" mode
+		if ( $this->onlyRenameGlobalUsers ) {
+			return $uncycloUser;
+		}
+
+		$user = clone $uncycloUser;
+		$newName = self::PREFIX_RENAME_UNCYCLO . $user->getName();
+
+		// check user name conflicts
+		if ( $this->getGlobalUserByName( $newName ) ) {
+			$newName .= '-1';
```

Comment:
what if this name also exists? it's worth throwing an exception here in such case